libfreerdp/core/gcc.c

```diff
@@ -2387,15 +2389,31 @@ BOOL gcc_read_server_multitransport_channel_data(wStream* s, rdpMcs* mcs)
 	if (!Stream_CheckAndLogRequiredLength(TAG, s, 4))
 		return FALSE;
 
-	Stream_Read_UINT32(s, settings->MultitransportFlags); /* flags, merge with client setting */
+	Stream_Read_UINT32(s, remoteFlags);
+	settings->MultitransportFlags &= remoteFlags; /* merge with client setting */
 	return TRUE;
 }
 
 BOOL gcc_write_server_multitransport_channel_data(wStream* s, const rdpMcs* mcs)
 {
-	UINT32 flags = 0;
-	if (!gcc_write_user_data_header(s, SC_MULTITRANSPORT, 8))
-		return FALSE;
-	Stream_Write_UINT32(s, flags); /* flags (4 bytes) */
+	WINPR_ASSERT(s);
+	WINPR_ASSERT(mcs);
+
+	rdpContext* context;
+	rdpSettings* settings;
+
+	context = transport_get_context(mcs->transport);
+	WINPR_ASSERT(context);
+
+	settings = context->settings;
+	WINPR_ASSERT(settings);
+
+	if (settings->SupportMultitransport && settings->MultitransportFlags)
```

Comment:
Might be a good idea to move this check out of this function (e.g. keep the `write` and don´t do an `maybe write`)